src/Security/MemberAuthenticator/MemberAuthenticator.php

```diff
@@ -206,7 +207,16 @@ protected function recordLoginAttempt($data, HTTPRequest $request, $member, $suc
 
         $attempt->Email = $email;
         $attempt->IP = $request->getIP();
+
+        if ($member) {
+            $member->extend('updateLoginAttempt', $attempt, $data, $request);
+        }
+
+        $this->invokeWithExtensions('updateLoginAttempt', $attempt, $data, $request);
```

Comment:
Isn't this more or less the same as the previous line?

Reply:
Agreed - IMO it’s also nicer to only have the extension hook invoked on one class. You should just be able to add the `$member` variable to the second call right?

Reply:
Thanks @robbieaverill @kinglozzer. I neglected to consider that member is accessible via `$attempt->Member()`

Reply:
I chose to remove the earlier `extend()` because in my use case I got the most value from working from an extension of the Authenticator. Let me know if you think having the extension method available to an extension of `Member` would be better

Reply:
I think what you’ve left is better